.travis.yml

```
@@ -0,0 +1,37 @@
+---
+dist: trusty
+os: linux
+language: ruby
+cache: bundler
+rvm:
+  - 2.4.5
+  - 2.5.8
+  - 2.6.5
+  - 2.7.1
+  - ruby-head
+gemfile:
+  - gemfiles/rails_4.gemfile
+  - gemfiles/rails_5.gemfile
+  - gemfiles/rails_6.gemfile
+jobs:
+  fast_finish: false
+  exclude:
+    - gemfile: gemfiles/rails_5.gemfile
+      rvm: 2.4.5
+    - gemfile: gemfiles/rails_6.gemfile
+      rvm: 2.4.5
+  allow_failures:
+    - rvm: ruby-head
+install:
+  - gem env
+  - gem list bundler
+  - rm -f /home/travis/.rvm/rubies/ruby-2.*/lib/ruby/gems/2.*/specifications/default/bundler-2.*.gemspec
+  - echo y | rvm @global do gem install bundler -v 1.17.3 --force --default
+  - gem install bundler -v 1.17.3 --force --default
+  - gem install bundler -v 1.17.3
+  - gem list bundler
+  - bundle --version
+  - bundle install --jobs=3 --retry=3 --path=${BUNDLE_PATH:-vendor/bundle}
```

By moving the installation of the proper bundle version into the `before_install` section the `bundle install` step is unnecessary. It should simplify a lot of this to
```suggestion
before_install:
  - gem install bundler -v 1.17.3 --force --default
```

---

The `rm -f` part is necessary. But I am moving up to `before_install` and removing some of the debugging output.